src/t_hash.c

```diff
@@ -858,7 +858,8 @@ static void addHashIteratorCursorToReply(client *c, hashTypeIterator *hi, int wh
 void genericHgetallCommand(client *c, int flags) {
     robj *o;
     hashTypeIterator *hi;
-    int length, count = 0;
+    long length;
+    long long count = 0;
```

Comment:
unclear why we need a long long here, wouldn't unsigned long be sufficient?

Reply:
Because count may be twice as long as length.

Reply:
yeah, but length is signed, so the unsigned variant supports 2x + 1 of the signed width. So maybe length should be unsigned?

Reply:
yeah, I am checking the code, set the max of the key to unsigned long, and set greater than the max value to unsigned long long.